chia/rpc/full_node_rpc_api.py

```diff
@@ -397,6 +404,60 @@ async def get_block_records(self, request: Dict) -> Optional[Dict]:
             records.append(record)
         return {"block_records": records}
 
+    async def get_block_spends(self, request: Dict) -> Optional[Dict]:
+        if "header_hash" not in request:
+            raise ValueError("No header_hash in request")
+        header_hash = bytes32.from_hexstr(request["header_hash"])
+        full_block: Optional[FullBlock] = await self.service.block_store.get_full_block(header_hash)
+        if full_block is None:
+            raise ValueError(f"Block {header_hash.hex()} not found")
+
+        ref_list = full_block.transactions_generator_ref_list
+        tx_info = full_block.transactions_info
+        generator_program = full_block.transactions_generator
+        height = full_block.reward_chain_block.height
+        spends: List[CoinSpend] = []
+        if tx_info and generator_program:
+            cost = tx_info.cost
+            generator_args: List[SerializedProgram] = []
+            for height in ref_list:
```

Comment:
There is a method for this, `self.service.blockchain.get_block_generator`, so you can decrease the amount of code here